packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/components/webVitalsUtils.ts

```diff
@@ -0,0 +1,252 @@
+}
+
+export const clsDataAverage = (webVitalData: WebVitalsResponseItem[]) => {
+    const clsData: number[] = [];
+    if (webVitalData) {
+        for (const item of webVitalData) {
+            if (item.name === "CLS") {
+                clsData.push(item.value);
+            }
+        }
+    }
+    return calculateClsAverage(clsData);
+};
+
+export function calculateClsScoreText(cls: number): string {
+    if (cls <= 0.1) {
+        return "Excellent"; // Excellent
+    }if (cls <= 0.25) {
+        return "Good"; // Good to Excellent
+    }if (cls <= 0.5) {
+        return "Fair"; // Fair to Good
+    }
+    return "Poor"; // Poor to Fair
+}
+
+export function calculateClsScorePercent(cls: number): number {
+    if (cls <= 0.1) {
+        return 100; // Excellent
+    }if (cls <= 0.25) {
+        return Math.round(100 - ((cls - 0.1) / (0.25 - 0.1)) * 25); // Good to Excellent
+    }if (cls <= 0.5) {
+        return Math.round(75 - ((cls - 0.25) / (0.5 - 0.25)) * 25); // Fair to Good
+    }
+    return Math.round(50 - ((cls - 0.5) / (1 - 0.5)) * 50); // Poor to Fair
+}
+
+export const progressBarClsColor = (clsData: number) => {
+    if (clsData <= 0.25) {
+        return "green";
+    }
+    if (clsData > 0.25 && clsData <= 0.5) {
+        return "yellow";
+    }
+    return "red";
+}
+
+export const progressBarClsTrackColor = (clsData: number) => {
+    if (clsData <= 0.25) {
+        return "yellow";
+    }
+    if (clsData > 0.25) {
+        return "red";
+    }
+    return "red";
+}
+
+export const clsTextColor = (clsData: number) => {
+    if (clsData <= 0.25) {
+        return "green";
+    }
+    if (clsData > 0.25 && clsData <= 0.5) {
+        return "yellow";
+    }
+    return "red";
+}
+
+// LCP Calculations
+export function calculateLcpAverage(lcpValues: number[]): number {
+
+    const sum = lcpValues.reduce((acc, curr) => acc + curr, 0);
+    const average = sum / lcpValues.length;
+    
+    // Round average to two decimal places
+    return Math.floor(Math.round(average * 100) / 100);
+}
+
+export const lcpDataAverage = (webVitalData: WebVitalsResponseItem[]) => {
+    const lcpData: number[] = [];
+    if (webVitalData) {
+        for (const item of webVitalData) {
+            if (item.name === "LCP") {
+                lcpData.push(item.value);
+            }
+        }
+    }
+    return calculateLcpAverage(lcpData);
+};
+
+export function calculateLcpScoreText(lcp: number): string {
+    if (msToSeconds(lcp) <= 2) {
+        return "Excellent"; // Excellent
+    }if (msToSeconds(lcp) <= 4) {
+        return "Good"; // Good to Excellent
+    }if (msToSeconds(lcp) <= 6) {
+        return "Fair"; // Fair to Good
+    }
+    return "Poor"; // Poor to Fair
+}
+
+export function calculateLcpScorePercent(lcp: number): number {
+    if (msToSeconds(lcp) <= 2) {
+        return 100; // Excellent
+    }if (msToSeconds(lcp) <= 4) {
+        return Math.round(100 - ((lcp - 2) / (4 - 2)) * 50); // Good to Excellent
```

Comment:
Same for these calculations

Reply:
https://web.dev/articles/lcp

Reply:
[2e5f04a](https://github.com/astrolicious/studiocms/pull/125/commits/2e5f04a92da69eb44ba45385409c4d99014b2878) comments